dao/authentication_dao.go

```diff
@@ -124,6 +124,10 @@ func (a *authenticationDaoImpl) ListForSource(sourceID int64, _, _ int, _ []util
 		}
 	}
 
+	if len(out) == 0 {
+		return nil, 0, util.NewErrNotFound("source")
+	}
+
```

Comment:
I think that this code block is not needed, since we are in a `List*` function. When you are querying for a collection in a REST API, having the empty collection "[]" as the indicator that there are no elements is a valid response. There is no need to return a 404 in that case.

Reply:
Yes, you're right. My idea was to handle scenario when source doesn't exists. Because endpoint is `GET /api/sources/v3.1/sources/{id}/authentications`. I prepared second version, where I try to find out if source exists or not and then continue to get list of authentications.